clientconn.go

```diff
@@ -444,6 +444,19 @@ func (csm *connectivityStateManager) getNotifyChan() <-chan struct{} {
 	return csm.notifyChan
 }
 
+// ClientConnInterface defines the functions clients need to perform unary and
+// streaming RPCs.  It is implemented by ClientConn.
```

Comment:
Should we also talk about the wrappers should return a concrete struct that implements this interface, not this interface directly?

Reply:
I started writing something about this but it felt like generic golang best practice, so I removed it.  How about something like "it is only intended to be referenced by generated code"?  (Updated)

Reply:
Seems like a grpc-internal detail.

I think that any code interacting with grpc conns should use the interface, as I think user code shouldn't need any of the concrete methods on ClientConn like GetState, GetMethodConfig. (Maybe I'm totally wrong about that?)

Reply:
The sole intention of this interface is to enable the generated code to accept an interface (as function parameters should typically be where possible).  Users should not otherwise reference or interact with this type; they should use the concrete type instead.